Add Iterable examples (#118)
radarlabs/documentation: docs/integrations/iterable.mdx

Radar.setUserId(userId)
```

## Example use cases

> You can use Radar and Iterable together to power strong customer engagement use cases. 

Reviewer: Radar and Iterable can be leveraged to create powerful and flexible location based marketing campaigns.

### Send a personalized push notification when user is near a store
You'll need access to both Radar and Iterable to build this messaging campaign. In this campaign, we'll set up a simple push notification based on a Radar geofence entry event.

1. First, ensure your [Geofences](/geofences) are configured in Radar. For a "nearby" use case, we typically suggest something like a 500 meter radius. You may also want to take advantage of *isochrone* (time-based) geofences, which would allow you to define, for example, a geofence that shows all points within a 10 minute walk from your target location.
2. Note the **tag** for the nearby geofences you've set up in step one. You can find your geofence tags in Radar by viewing your [geofences](https://radar.com/dashboard/geofences).
3. In Iterable, navigate to [Journeys](https://app.iterable.com/workflows) and click **+ New journey**. For more on Journeys in Iterable, please refer to [Iterable's documentation](https://support.iterable.com/hc/en-us/articles/4405798856212-Introduction-to-Journeys-).

> 4. Iterable will start your Journey with a "start" node by default. Double click this node and change the **Received Api Trigger** dropdown value to **Triggered Custom Event**. Under *Event name*, start typing *Radar* and you should see a list of available events begin to auto-populate. Select *radarGeofenceEntered*.

Reviewer: Iterable will "create" a journey?

5. Change the **Use Filter** option to *Yes*.
6. Where it says **Select a property**, start typing *RadarGeofence* and look for the *radarGeofenceEntered.geofenceTag* property. Set this property equal to the tag you set up in step one and verified in step two.
7. Click **Update Node**.
8. Add a new node via Iterable's drag and drop functionality. For this example, drag in the **Send Push** node, then connect the *Start* node with the *Send Push* node.
9. Now, we'll need to create an [Iterable Template](https://support.iterable.com/hc/en-us/articles/205480315-Introduction-to-Templates-) to create a personalized push to be used for this campaign. Navigate to Content -> Templates, and click **+ New template** -> **Push** -> **Create template** (giving it a name as needed).

> 10. We're going to customize the title of the Template using data from Radar so your message can say "You're near {{Store name}}", with {{Store name}} representing the name of the nearby store whose geofence the user has entered. In the **Title** text box, enter **{{radarGeofenceDescriptions}}**. This syntax is called *Handlebar* notation. You can read more about it in Iterable's [documentation](https://support.iterable.com/hc/en-us/articles/205480365-Using-Merge-Parameters-and-Handlebars-Logic-for-Personalization).

Reviewer: should we explain the description use higher up?

11. Add in a Push message, then click **Save Template**.
12. Back in the *Journey*, double click on the **Send mobile push** node and change the template to the one you just created.

> That's it!

Reviewer: Should we reference simulation to test the push and link to that section?

1. First, let's configure [Geofences](/geofences) in Radar. Read more about creating geofences [here](/geofences#create-geofences). Set up isochrone geofences with a 10 minute drive time.
2. Note the **tag** for the 10 minute drive geofences you've set up in step one. You can find your geofence tags in Radar by viewing your [geofences](https://radar.com/dashboard/geofences).
3. In Iterable, navigate to [Journeys](https://app.iterable.com/workflows) and click **+ New journey**. For more on Journeys in Iterable, please refer to [Iterable's documentation](https://support.iterable.com/hc/en-us/articles/4405798856212-Introduction-to-Journeys-).
4. Iterable will start your Journey with a "start" node by default. Double click this node and change the **Received Api Trigger** dropdown value to **Opened An Email**. Select the campaign and filter as desired. Click **Update Node**.
5. Now, we'll add a new node that filters this initial trigger down based on Radar data. Under *Filters* on the left sidebar of Iterable's Journey builder, drag **Fields Match** in below the Start node.

> 6. Double click on this new **Filter users** node. Where it says *Choose a property*, select *Custom Event*. Type in *radarGeofenceEntered.geofenceTag*, and set it equal to the tag you created in step one and verified in step two. Click **Update Node**.

Reviewer: don't we want this to be a contact property, not the event?

7. Connect the *Start* node with the *Filter users* node.
8. Now, we'll need to create an [Iterable Template](https://support.iterable.com/hc/en-us/articles/205480315-Introduction-to-Templates-) to create a personalized push to be used for this campaign. Navigate to Content -> Templates, and click **+ New template** -> **Push** -> **Create template** (giving it a name as needed).

> 10. We're going to customize the title of the Template using data from Radar so your message can say "You're near {{Store name}}", with {{Store name}} representing the name of the nearby store whose geofence the user has entered. In the **Title** text box, enter **{{radarGeofenceDescriptions}}**. This syntax is called *Handlebar* notation. You can read more about it in Iterable's [documentation](https://support.iterable.com/hc/en-us/articles/205480365-Using-Merge-Parameters-and-Handlebars-Logic-for-Personalization).

Reviewer: if we are going off contact properties, are we going to have issues with this? The array of descriptions would not be sorted in an order.

Author: Yeah - you're right. I'm going to take the customization out for this one. I don't think it's as vital. A diff. tutorial showing customization based on events would be good, potentially

Reviewer: it is pretty valuable

Author: good call. will do that now

Author: ah wait - I already have that for the first example, so I think we're set

![image](https://user-images.githubusercontent.com/1242056/160176013-699d0d03-eabb-4c10-a02e-cbd4571eadbf.png)

Radar.setUserId(userId)
```

## Example use cases

> Radar and Iterable can be leveraged to create powerful and flexible location based marketing campaigns.

Reviewer 2: ```suggestion
Learn how to create powerful location-based messaging campaigns using Radar and Iterable.
```

> ### Send a personalized push notification when user is near a store

Reviewer 2: We may want to group this into sections by subheadings:

**Configure your geofences in Radar for your campaign**
- Step 1
- Step 2

**Create a new Journey in Iterable**
- Step 3
- Step 4
- Step 5
- Step 6
- Step 7

**Set up a push notification with a new Template**
- Step 8
- Step 9
- Step 10
- Step 11
- Step 12

**Testing your Iterable campaign**
Test section

You'll need access to both Radar and Iterable to build this messaging campaign. In this campaign, we'll set up a simple push notification based on a Radar geofence entry event.

> 1. First, ensure your [Geofences](/geofences) are configured in Radar. For a "nearby" use case, we typically suggest something like a 500 meter radius. You may also want to take advantage of *isochrone* (time-based) geofences, which would allow you to define, for example, a geofence that shows all points within a 10 minute walk from your target location.

Reviewer: "a geofence with boundaries set to all points" or something to that effect (looking for a different verb than "shows")

> 2. Note the **tag** for the nearby geofences you've set up in step one. You can find your geofence tags in Radar by viewing your [geofences](https://radar.com/dashboard/geofences).

Reviewer: do we need this a separate step? could just be a note on step 1

Author: It makes sense to me as a step (albeit one that folks may just skip since they've already set up geofences) but open to changing if you have a strong opinion. just let me know your preference.

3. In Iterable, navigate to [Journeys](https://app.iterable.com/workflows) and click **+ New journey**. For more on Journeys in Iterable, please refer to [Iterable's documentation](https://support.iterable.com/hc/en-us/articles/4405798856212-Introduction-to-Journeys-).
4. New Journeys in Iterable have a "start" node by default. Double click this node and change the **Received Api Trigger** dropdown value to **Triggered Custom Event**. Under *Event name*, start typing *Radar* and you should see a list of available events begin to auto-populate. Select *radarGeofenceEntered*.
5. Change the **Use Filter** option to *Yes*.

> 6. Where it says **Select a property**, start typing *RadarGeofence* and look for the *radarGeofenceEntered.geofenceTag* property. Set this property equal to the tag you set up in step one and verified in step two.

Reviewer: missed this - should "start typing RadarGeofence" be "radarGeofence"

Author: it actually doesn't matter -- Iterable is not case sensitive in the event search. I went with RadarGeofence just because it seems simpler/what someone would probably type, but open to changing if you think I should

7. Click **Update Node**.
8. Add a new node via Iterable's drag and drop functionality. For this example, drag in the **Send Push** node, then connect the *Start* node with the *Send Push* node.
9. Now, we'll need to create an [Iterable Template](https://support.iterable.com/hc/en-us/articles/205480315-Introduction-to-Templates-) to create a personalized push to be used for this campaign. Navigate to Content -> Templates, and click **+ New template** -> **Push** -> **Create template** (giving it a name as needed).

> 10. We're going to customize the title of the Template using data from Radar so your message can say "You're near {{Store name}}", with {{Store name}} representing the name of the nearby store whose geofence the user has entered. In the **Title** text box, enter **{{radarGeofenceDescriptions}}**. This syntax is called *Handlebar* notation. You can read more about it in Iterable's [documentation](https://support.iterable.com/hc/en-us/articles/205480365-Using-Merge-Parameters-and-Handlebars-Logic-for-Personalization).

Author: yep, you're right

11. Add in a Push message, then click **Save Template**.
12. Back in the *Journey*, double click on the **Send mobile push** node and change the template to the one you just created.

To test this campaign, you can either trigger the Journey on a test app by moving physically into the geofence, *or* you can take advantage of the [Radar simulator](/dashboard#simulator). The simulator allows you to simulate a specific user as if they were moving throughout the world.

> Navigate to the **Users** page in Radar and select a user who exists in Iterable and is linked via id. Then, click the *Simulate* button at the top of the page. Click around on the map and enter a geofence which will trigger the campaign you built in the steps above. If hooked up correctly, you should receive the relevant notification on your device.

Reviewer: linked via **Radar user id** (just want to be specific

### Filter users in an Iterable Journey whose last known location is X minutes away from a store

> You'll need access to both Radar and Iterable to build this messaging campaign. In this example, we'll set up a push campaign that is sent to users who have opened an email associated with a specific Iterable campaign **and** are within 10 minutes drive of a store. We can add isochrone (time-based) geofences (via import, the UI, or the API) in Radar to support this use case.

Reviewer 2: Same feedback on splitting it into sub-sections as above

1. First, let's configure [Geofences](/geofences) in Radar. Read more about creating geofences [here](/geofences#create-geofences). Set up isochrone geofences with a 10 minute drive time.
2. Note the **tag** for the 10 minute drive geofences you've set up in step one. You can find your geofence tags in Radar by viewing your [geofences](https://radar.com/dashboard/geofences).
3. In Iterable, navigate to [Journeys](https://app.iterable.com/workflows) and click **+ New journey**. For more on Journeys in Iterable, please refer to [Iterable's documentation](https://support.iterable.com/hc/en-us/articles/4405798856212-Introduction-to-Journeys-).
4. Iterable will start your Journey with a "start" node by default. Double click this node and change the **Received Api Trigger** dropdown value to **Opened An Email**. Select the campaign and filter as desired. Click **Update Node**.
5. Now, we'll add a new node that filters this initial trigger down based on Radar data. Under *Filters* on the left sidebar of Iterable's Journey builder, drag **Fields Match** in below the Start node.

> 6. Double click on this new **Filter users** node. Where it says *Choose a property*, select *Contact Property*. Type in *radarGeofenceTags*, and set it equal to the tag you created in step one and verified in step two. Click **Update Node**.

Reviewer: "set it equal" should be a contains match in case you are in other other geofences (it is an array of current geofences the device is in)
